### ISSUE TITLE
Commands flags with quoted parameters lose flags.

### ISSUE THREAD

**Issue by joshriverscambia2019** [nat-n/poethepoet#177]
When you define a command task with a flag parameter with an equals sign and a quoted operand, the flag is omitted from the command execution.

For  example, the task:

```
foo = "ls --color='always'"
```

Will output:

```
% poetry run poe foo
Poe => ls always
ls: always: No such file or directory
```

While one would expect:

```
% poetry run poe foo
Poe => ls --color=always
README.md      pyproject.toml
```

This problem began with v0.23.0. It seems likely that the issue is that the `resolve_command_tokens` function is losing data in `token_parts` variable when it is re-initialized within [this loop](https://github.com/nat-n/poethepoet/blob/main/poethepoet/helpers/command/__init__.py#L59).

**Comment by nat-n** [comment]
Hi @joshriverscambia2019 

Thanks for the feedback :) yep that's a bug in the new cmd parsing logic! I'll get a fix for that out soon :)

**Comment by nat-n** [comment]
Thanks again for reporting this bug. It's now fixed with [v0.24.2](https://pypi.org/project/poethepoet/0.24.2/) 🚀
